internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -976,3 +997,77 @@ func (r *PersistentVolumeClaimReconciler) getScheduleFromPVC(
 
 	return ""
 }
+
+// checkAnnotationForValue checks if the given object has the specified annotation
+// with the expected value.
+func checkAnnotationForValue(obj metav1.Object, key, expected string) bool {
+	if val, ok := obj.GetAnnotations()[key]; ok && val == expected {
+		return true
+	}
+	return false
+}
+
+// hasValidStorageClassName checks if the provided PersistentVolumeClaim has a non-empty StorageClassName.
+func hasValidStorageClassName(pvc *corev1.PersistentVolumeClaim) bool {
+	return pvc.Spec.StorageClassName != nil && len(*pvc.Spec.StorageClassName) > 0
+}
+
+// checkDisabledByAnnotation checks if the given object has an annotation
+// that disables the functionality represented by the provided annotationKey.
+func (r *PersistentVolumeClaimReconciler) checkDisabledByAnnotation(
+	ctx context.Context,
+	logger *logr.Logger,
+	pvc *corev1.PersistentVolumeClaim,
+	annotationKey string) (bool, error) {
+	isDisabledOnSC := func() (bool, error) {
+		// Not an error, static PVs
+		if !hasValidStorageClassName(pvc) {
+			return false, nil
+		}
+
+		storageClassName := *pvc.Spec.StorageClassName
+		storageClass := &storagev1.StorageClass{}
+
+		err := r.Client.Get(ctx, client.ObjectKey{Name: storageClassName}, storageClass)
+		if err != nil {
+			logger.Error(err, "Failed to get StorageClass", "StorageClass", storageClassName)
+			return false, err
+		}
+
+		return checkAnnotationForValue(storageClass, annotationKey, "false"), nil
+	}
+
+	if r.SchedulePrecedence == util.ScheduleSCOnly {
+		disabled, err := isDisabledOnSC()
+		if err != nil {
+			return false, err
+		}
+
+		return disabled, nil
+	}
```

Comment:
wont we need to move this check to line 1028? where we check the Precedence first?